### PR TITLE
ADEN-11314 AdEngine stops working when the page is loaded in an iframe

### PR DESCRIPTION
**Pull request by szymon-nawrocki**
### Ticket
https://fandom.atlassian.net/browse/ADEN-11314

### CODE REVIEW

**Review by nandy-andy — APPROVED** [review]

File: platforms/shared/consent/tracking-opt-in-wrapper.ts [hunk 1]
```diff
@@ -26,7 +26,13 @@ class TrackingOptInWrapper {
 			communicationService.on(eventsRepository.AD_ENGINE_CONSENT_READY, (payload) => {
```
Comment (szymon-nawrocki) [inline platforms/shared/consent/tracking-opt-in-wrapper.ts:26]: The communicationService does not work properly when site is run inside iframe, so the `AD_ENGINE_CONSENT_READY` most of the time is not send from the backend, so I've added the lines below to go around this obstacle.
I know that it is far from perfect solution, but we need it only for a week or two to collect a data of how often AdEngine is being loaded in iframe, and I couldn't think of a better idea of how to do it.

File: platforms/shared/consent/tracking-opt-in-wrapper.ts [hunk 2]
```diff
@@ -26,7 +26,13 @@ class TrackingOptInWrapper {
...
 				this.setConsents(payload);
 				resolve();
+				return;
 			});
+
+			const pageInIframe = document.documentElement.classList.contains('in-iframe');
```
Comment (szymon-nawrocki) [inline platforms/shared/consent/tracking-opt-in-wrapper.ts:32]: So I've added an additional promise resolve possibility, which is:
if `AD_ENGINE_CONSENT_READY` event is not sent from the backend -> check if site is loaded in iframe -> if it is -> resolve the promise

@nandy-andy
Do you think this solution could fly for a week or two, to collect some data of how often AdEngine is being loaded in iframe? Then we could probably remove it.

**Comment by szymon-nawrocki** [comment]
Reopening PR due to closing by mistake

File: platforms/shared/consent/tracking-opt-in-wrapper.ts (hunk 2 above)
Comment (nandy-andy) [inline platforms/shared/consent/tracking-opt-in-wrapper.ts:32]: how about `r/pageInIframe/isPageInIframe/`? :)

File: platforms/shared/consent/tracking-opt-in-wrapper.ts [hunk 3]
```diff
@@ -26,7 +26,13 @@ class TrackingOptInWrapper {
...
+			if (pageInIframe) {
+				resolve();
```
Comment (nandy-andy) [inline platforms/shared/consent/tracking-opt-in-wrapper.ts:34]: Don't we need to use `return` here as well as we do in line no. 29? Or maybe we don't need it in line no. 29 either? Or maybe we need to return `resolve();`? 🤔

File: platforms/shared/setup/iframe-detector.setup.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { DiProcess } from '@wikia/ad-engine';
+import { Injectable } from '@wikia/dependency-injection';
+import { PageTracker } from '../tracking/page-tracker';
+import { DataWarehouseTracker } from '../tracking/data-warehouse';
+
+@Injectable()
+export class IframeDetectorSetup implements DiProcess {
+	execute(): void {
+		const elementFound = document.documentElement.classList.contains('in-iframe');
```
Comment (nandy-andy) [inline platforms/shared/setup/iframe-detector.setup.ts:9]: element or class found? ;-)

**Pull request merged**